### ISSUE TITLE
AngleSharp - Share cookie over all request

### ISSUE THREAD

**Issue by IHNEL** [AngleSharp/AngleSharp#365]
I am really new with AngleSharp. I want to pull some pages behind login form. The first page right after login form works fine, but when I call loadAsync() to load second page, it ask me to login again. I am not sure, but I think the cookie was through away, Anybody meet this case before? Please help!.

since I can't publish the site I am working with, so I use Facebook.com as a sample:

``````
 ```
``````

> mConfig = Configuration.Default.WithDefaultLoader().WithCookies().WithLocaleBasedEncoding();
>      mContext = BrowsingContext.New(mConfig);
>      document = await mContext.OpenAsync("http://facebook.com");
> 
> ```
>  IHtmlInputElement email = document.QuerySelector("input[name~='email']") as       IHtmlInputElement;
>  IHtmlInputElement pass = document.QuerySelector("input[name~='pass']") as IHtmlInputElement;
>  IHtmlInputElement btnLogin = document.QuerySelector("input[name~='login'][type~='submit']") as IHtmlInputElement;
>  IHtmlFormElement loginForm = document.QuerySelector("form#login_form") as IHtmlFormElement;
>  var elements = document.QuerySelectorAll("input");
> 
>  if (btnLogin != null && email != null && pass != null)
>  {
>      email.Value = "testemail@gmail.com";
>      pass.Value = "secret";
>      document = await loginForm.SubmitAsync();
> 
>     //Open second page, Facebook ask to login again.
>      document = await context.OpenAsync("https://www.facebook.com/jenny.san.94/posts/1213295672022614");
>  }
> ```

```
```

**Comment by FlorianRappl** [comment]
I think this is fixed already (see #274 #280), however, not available in the latest version on Nuget (v0.9.6). I am getting v0.9.7 ready, but it has been postponed multiple times (my apologizes).

You could either build from the sources (`devel` branch) or wait a couple more days (in best case tonight, in the worst I guess Monday next week) before v0.9.7 is released.

Hope this helps!

**Comment by IHNEL** [comment]
I confirm that the latest source code on devel branch fixed the issue. Thank you so much.

**Comment by FlorianRappl** [comment]
Glad to hear that. Hoping to get v0.9.7 out soon!

**Comment by abozanona** [comment]
This problem isn't solved in **v0.9.10**

I'm coding Xamarin.Forms application.

**Comment by abozanona** [comment]
I temporarily solved this problem by
 - I extending `HttpRequester`  to `CustomHttpRequester` https://github.com/AngleSharp/AngleSharp/blob/5139da94d7dfece420c02f000b3d5729168242af/src/AngleSharp/Network/Default/HttpRequester.cs#L18
 - Replaced the cookies separator to `,` instead of `;` as mentioned in #385 
 - I had to make one more change to make it work, I made the [`CookiesContainer`](https://github.com/AngleSharp/AngleSharp/blob/5139da94d7dfece420c02f000b3d5729168242af/src/AngleSharp/Network/Default/HttpRequester.cs#L113) static, and checked if it equals none before [creating an instance of it](https://github.com/AngleSharp/AngleSharp/blob/5139da94d7dfece420c02f000b3d5729168242af/src/AngleSharp/Network/Default/HttpRequester.cs#L121)

My code now looks like this
 
```
static CookieContainer _cookies;
readonly HttpWebRequest _http;
readonly IRequest _request;
readonly Byte[] _buffer;

public RequestState(IRequest request, IDictionary<String, String> headers)
{
    string cookieHeader;
    request.Headers.TryGetValue(HeaderNames.Cookie, out cookieHeader);
    if(_cookies==null)
        _cookies = new CookieContainer();
    _request = request;
    _http = WebRequest.Create(request.Address) as HttpWebRequest;
```

I noticed that every time a new request is created, a new `RequestState` is created, which replaces the current `CookieContainer` with a new one, so I had to make the `CookieContainer` static.

**notice that** Making `_cookies` static will share it among all other instance, which is something I need in my project but not necessarily in other projects who might want the `CookieContainer` be separated for every object.
